File: platform-server/app/schemas/system_info.py
"""
System Information Schemas
"""
from typing import Dict, Optional
from app.schemas.base import BaseSchema


class K8sInfo(BaseSchema):
    """Kubernetes environment information"""
    pod_name: str
    node_name: str
    namespace: str


class DatabaseInfo(BaseSchema):
    """Desensitized database connection information"""
    host: str
    port: int
    user: str
    database: str
    adminer_url: Optional[str] = None


class RedisInfo(BaseSchema):
    """Desensitized redis connection information"""
    host: str
    port: int
    db: int
    insight_url: Optional[str] = None


class ServiceHealth(BaseSchema):
    """Service health status"""
    status: str
    version: Optional[str] = None
    last_check: Optional[str] = None


class SystemInfoResponse(BaseSchema):
    """System information response"""
    app_env: str
    k8s: K8sInfo
    database: DatabaseInfo
    redis: RedisInfo
    services: Dict[str, ServiceHealth]

File: platform-console/apps/raap-admin/src/api/core/system-info.ts
import { requestClient } from '#/api/request';

export namespace SystemInfoApi {
  export interface K8sInfo {
    pod_name: string;
    node_name: string;
    namespace: string;
  }

  export interface DatabaseInfo {
    host: string;
    port: number;
    user: string;
    database: string;
  }

  export interface RedisInfo {
    host: string;
    port: number;
    db: number;
    insight_url?: string;
  }

  export interface ServiceHealth {
    status: string;
    version?: string;
    last_check?: string;
  }

  export interface SystemInfoResult {
    app_env: string;
    k8s: K8sInfo;
    database: DatabaseInfo;
    redis: RedisInfo;
    services: Record<string, ServiceHealth>;
  }
}

/**
 * 获取系统信息
 */
export async function getSystemInfoApi() {
  return requestClient.get<SystemInfoApi.SystemInfoResult>('/v1/system/info');
}
